### ISSUE TITLE
Feature request: PreventCrossDataCenterIntermediateMasterFailover/PreventCrossRegionIntermediateMasterFailover

### ISSUE THREAD

**Issue by yozhovich** [openark/orchestrator#928]
I sometimes come up against the following situation: if IntermediateMaster fails, orchestrator selects a candidate from the other datacenter/region. I have to solve the problem using hooks, by setting the promotion rule to must_not for these candidates and setting back the promotion rule neutral for them after a failover.
 
Orchestrator has two usefull settings:
PreventCrossRegionMasterFailover
PreventCrossDataCenterMasterFailover

Unfortunately, they do not affect to IntermediateMaster
It would be also great to have
PreventCrossRegionIntermediateMasterFailover
PreventCrossDataCenterIntermediateMasterFailover

I think that these settings would be very useful for specialists using InetmediateMasters in the crossdatacenter and crossregion topologies.
 
What do you think about this?
Thank you in advance!

**Comment by shlomi-noach** [comment]
I think this makes sense -- but I'd also like to learn more on the situation. As it is, `orchestrator` strongly rpefers failing over an intermediate master within the same datacenter, and only promote an external-DC server if it is unable to promote one from the same DC.

In which case, do you wish `orchestrator` to completely fail the promotion? Otherwise, what is the behavior you see?

**Comment by assaggaf** [comment]
@shlomi-noach I also agree with the request here. In my case, yes the promotion should fail if it can not promote an intermediate master from the same DC/region. Which, according to my understanding form the code, it would next attempt regrouping and promote a child. That would be the desired behaviour when these two options are set:

PreventCrossRegionIntermediateMasterFailover
PreventCrossDataCenterIntermediateMasterFailover
